ajustando metodo is_valid como único

antoniamaia opened the pull request:
## Descrição
Fiz todas as alterações, mas ainda dá erro 😢 deixando em draft pra vc ver onde cheguei 😄 

## Mudanças Propostas

## Checklist de Revisão
<!--- Marque as caixas que se aplicam. Você pode deixar caixas desmarcadas se elas não se aplicarem.-->

- [ ] Eu li o [Contributing.md](https://github.com/brazilian-utils/brutils-python/blob/main/CONTRIBUTING.md)
- [ ] Os testes foram adicionados ou atualizados para refletir as mudanças (se aplicável).
- [ ] Foi adicionada uma entrada no changelog / Meu PR não necessita de uma nova entrada no changelog.
- [ ] A [documentação](https://github.com/brazilian-utils/brutils-python/blob/main/README.md) em português foi atualizada ou criada, se necessário.
- [ ] Se feita a documentação, a atualização do [arquivo em inglês](https://github.com/brazilian-utils/brutils-python/blob/main/README_EN.md). <!---Permitido uso de Google Tradutor/ChatGPT. -->
- [ ] Eu documentei as minhas mudanças no código, adicionando docstrings e comentários. [Instruções](https://github.com/brazilian-utils/brutils-python/blob/main/CONTRIBUTING.md#8-fa%C3%A7a-as-suas-altera%C3%A7%C3%B5es)
- [ ] O código segue as diretrizes de estilo e padrões de codificação do projeto.
- [ ] Todos os testes passam. [Instruções](https://github.com/brazilian-utils/brutils-python/blob/main/CONTRIBUTING.md#testes)
- [ ] O Pull Request foi testado localmente. [Instruções](https://github.com/brazilian-utils/brutils-python/blob/main/CONTRIBUTING.md#7-execute-o-brutils-localmente)
- [ ] Não há conflitos de mesclagem.

## Comentários Adicionais (opcional)
<!--- Adicione qualquer informação adicional ou contexto que você acha importante para os revisores entenderem suas mudanças.-->

## Issue Relacionada
<!---Todos os PRs devem ter uma issue relacionada. Dessa forma, podemos garantir que ninguém perca tempo trabalhando em algo que não precisa ser feito. -->

Closes #259

cuducos commented on tests/test_license_plate.py:
```diff
@@ -1,26 +1,141 @@
+                "brutils.license_plate._is_valid_old_format", return_value=False
+            ) as mock__is_valid_old_format:
+                # When licence plate old format isn't valid, returns False
+                self.assertIs(is_valid("123456", "old_format"), False)
```
Na função você utilizou `"old format"` — com espaço ao invés de `_`. Por isso não chama o _mock_ como esperado aqui : )

cuducos commented on tests/test_license_plate.py:
```diff
@@ -1,26 +1,141 @@
+                "brutils.license_plate._is_valid_old_format"
+            ) as mock__is_valid_old_format:
+                # When license plate as mercosul is_valid, returns True
+                self.assertIs(is_valid("ABC4E67"), True)
```
Como a função `is_valid` retorna `_is_valid_old_format(…) or _is_valid_mercosul(…)`, se não retornar algo q se pareça com `False`, nunca vamos chegar no `_is_valid_mercosul(…)`.

Como vc não definiu o retorno do _mock_ de `_is_valid_old_format`, ele retorna um objeot `Mock()` que não se parece com `False`. Por isso nunca executamos a função do Mercosul como esperado : )

```suggestion
            with patch(
                "brutils.license_plate._is_valid_old_format", return_value=False
            ) as mock__is_valid_old_format:
                # When license plate as mercosul is_valid, returns True
                self.assertIs(is_valid("ABC4E67"), True)
```

antoniamaia committed:
Refatorando alguns erros

antoniamaia commented on tests/test_license_plate.py:
```diff
@@ -1,26 +1,145 @@
+
+                # Checks if function _is_valid_mercosul is called with the
+                # correct argument
+                # mock__is_valid_mercosul.assert_called_once_with("ABC4E67") <--
```
@camilamaia mock outra vez fundindo o meu motor 🤯 deixei essas coisas feias (<---) nas linhas comentadas que dão sempre o mesmo Assertion error.. quando é pra ser chamado, (e é chamado pq eu printei 💅🏼 ) ele diz que não foi chamado nenhuma vez, e o contrário a mesma coisa. Quando estava procurando documentação, em algum lugar (bem obcuro) disse que o mock pode ter um side effect quando usado muitas vezes, e utilizar essa função reset ajudaria. Para esse caso realmente funcionou, mas quis colocar tudo aqui pra vc acompanhar todo o rolê :)

antoniamaia commented on tests/test_license_plate.py:
```diff
@@ -1,26 +1,145 @@
+
+                # Checks if function _is_valid_mercosul is called with the
+                # correct argument
+                # mock__is_valid_mercosul.assert_called_once_with("ABC4E67") <--
```
@camilamaia aqui :)

antoniamaia commented on tests/test_license_plate.py:
```diff
@@ -1,26 +1,145 @@
+                mock__is_valid_mercosul.assert_called_once_with("ABC-123")
+                # Checks if function _is_valid_old_format is called with the
+                # correct argument
+                # mock__is_valid_old_format.assert_called_once_with("123-456")<--
```
@camilamaia aqui :)

dependabot[bot] committed:
build(deps-dev): bump identify from 2.5.31 to 2.5.32 (#275)

Bumps [identify](https://github.com/pre-commit/identify) from 2.5.31 to 2.5.32.
- [Commits](https://github.com/pre-commit/identify/compare/v2.5.31...v2.5.32)

---
updated-dependencies:
- dependency-name: identify
  dependency-type: indirect
  update-type: version-update:semver-patch
...

Signed-off-by: dependabot[bot] <support@github.com>
Co-authored-by: dependabot[bot] <49699333+dependabot[bot]@users.noreply.github.com>

camilamaia reviewed (commented):
Boa! Acho que com o pair que a gente fez, arrumando os mocks de um jeito mais claro, ficou mais suave para entender tb. My bad de ter criado tudo numa confusão só.
